bukkit/src/main/java/com/viaversion/viaversion/bukkit/providers/BukkitPickItemProvider.java

```diff
@@ -40,19 +39,31 @@
 import org.bukkit.inventory.PlayerInventory;
 import org.bukkit.inventory.meta.BlockStateMeta;
 
+import java.lang.reflect.Method;
+import java.util.UUID;
+
 public class BukkitPickItemProvider extends PickItemProvider {
     private static final BlockToItem BLOCK_TO_ITEM = BlockToItem.build();
     private static final GetStorageContents GET_STORAGE_CONTENT = GetStorageContents.build();
     private static final SetInHand SET_IN_HAND = SetInHand.build();
 
-    private static final boolean HAS_SPAWN_EGG_METHOD = PaperViaInjector.hasMethod(ItemFactory.class, "getSpawnEgg", EntityType.class);
+    private static final boolean HAS_SPAWN_EGG_METHOD = PaperViaInjector.hasMethod(ItemFactory.class, Material.class, "getSpawnEgg", EntityType.class);
     private static final double BLOCK_RANGE = 4.5 + 1;
     private static final double BLOCK_RANGE_SQUARED = BLOCK_RANGE * BLOCK_RANGE;
     private static final double ENTITY_RANGE = 3 + 3;
     private final ViaVersionPlugin plugin;
+    private Method getSpawnEggMethod;
 
     public BukkitPickItemProvider(final ViaVersionPlugin plugin) {
         this.plugin = plugin;
+
+        if (PaperViaInjector.hasMethod(ItemFactory.class, ItemStack.class, "getSpawnEgg", EntityType.class)) {
+            try {
+                getSpawnEggMethod = ItemFactory.class.getDeclaredMethod("getSpawnEgg", EntityType.class);
+            } catch (final ReflectiveOperationException e) {
+                throw new RuntimeException(e);
+            }
+        }
```

Comment:
This should be put into the static initializer and be called GET_SPAWN_EGG_ITEMSTACK_METHOD or something